statsmodels/doe/latin.py

```diff
@@ -0,0 +1,226 @@
+    niter : int
+        Number of iteration to perform.
+    force : bool
+        If `optimization`, force *basinhopping* optimization. Otherwise
+        grid search is used.
+    optimization : bool
+        Optimal design using global optimization or random generation of
+        `niter` samples.
+    Returns
+    -------
+    sample : array_like (n_samples, k_vars)
+        Optimal Latin hypercube Sampling.
+
+    References
+    ----------
+    [1] Damblin et al., "Numerical studies of space filling designs:
+    optimization of Latin Hypercube Samples and subprojection properties",
+    Journal of Simulation, 2013.
+
+    """
+    doe = start_design
+    if (bounds is None) and (doe is not None):
+            bounds = np.array([doe.min(axis=0), doe.max(axis=0)])
+    if optimization:
+        if doe is None:
+            doe = orthogonal_latin_hypercube(dim, n_sample, bounds)
+
+        def _perturb_doe(x, sample, bounds):
+            """Perturb the Design of Experiment.
+
+            Parameters
+            ----------
+            x : list of int
+                It is a list of:
+                    idx : int
+                        Index value of the components to compute
+            sample : array_like (n_samples, k_vars)
+                Sample to perturb.
+            bounds : tuple or array_like ([min, k_vars], [max, k_vars])
+                Desired range of transformed data. The transformation apply the
+                bounds on the sample and not the theoretical space, unit cube.
+                Thus min and max values of the sample will coincide with the
+                bounds.
+
+            Returns
+            -------
+            discrepancy : float
+                Centered discrepancy.
+
+            """
+            doe = copy.deepcopy(sample)
+            col, row_1, row_2 = np.round(x).astype(int)
+            doe[row_1, col], doe[row_2, col] = doe[row_2, col], doe[row_1, col]
+
+            return discrepancy(doe, bounds)
+
+        # Total number of possible design
+        complexity = dim * n_sample ** 2
+
+        if have_basinhopping and ((complexity > 1e6) or force):
+            bounds_optim = ([0, dim - 1], [0, n_sample - 1], [0, n_sample - 1])
+        else:
+            bounds_optim = (slice(0, dim - 1, 1), slice(0, n_sample - 1, 1),
+                            slice(0, n_sample - 1, 1))
+
+        for n in range(niter):
+            if have_basinhopping and ((complexity > 1e6) or force):
+                minimizer_kwargs = {"method": "L-BFGS-B",
+                                    "bounds": bounds_optim,
+                                    "args": (doe, bounds)}
+                optimum = basinhopping(_perturb_doe, [0, 0, 0], niter=100,
+                                       minimizer_kwargs=minimizer_kwargs).x
+            else:
+                optimum = brute(_perturb_doe, ranges=bounds_optim,
+                                finish=fmin, args=(doe, bounds))
```

Comment:
this still has `finish=fmin`
(just comparing to my local changes before switching branches)

Reply:
Oh yes, my mistake. I will fix this.